### PR TITLE
[MIG] purchase_request: Migrate to version 17.0

### PR DESCRIPTION
**Pull request by sonhd91**
Ref: BSRD-710

### CODE REVIEW

File: purchase_request/views/purchase_order_view.xml [hunk 1]
```diff
@@ -8,9 +8,9 @@
         <field name="inherit_id" ref="purchase.purchase_order_form" />
         <field name="arch" type="xml">
             <xpath expr="//field[@name='order_line']/tree" position="inside">
-                <field name="purchase_request_lines" invisible="1" />
+                <field name="purchase_request_lines" column_invisible="1" />
                 <button
-                    attrs="{'invisible': [('purchase_request_lines', '=', [])]}"
+                    column_invisible="purchase_request_lines == False"
```
Comment (youring) [inline purchase_request/views/purchase_order_view.xml:13]: invisible="not purchase_request_lines"

File: purchase_request/views/purchase_order_view.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+<odoo>
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="name">purchase.order.form</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/tree" position="inside">
+                <field name="purchase_request_lines" column_invisible="1" />
+                <button
+                    column_invisible="purchase_request_lines == False"
```
Comment (youring) [inline purchase_request/views/purchase_order_view.xml]:
```suggestion
                    invisible="not purchase_request_lines"
```

File: purchase_request/views/purchase_order_view.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    </record>
+    <record id="purchase_order_line_tree" model="ir.ui.view">
+        <field name="name">purchase.order.line.tree</field>
+        <field name="model">purchase.order.line</field>
+        <field name="inherit_id" ref="purchase.purchase_order_line_tree" />
+        <field name="arch" type="xml">
+            <xpath expr="//tree" position="inside">
+                <field name="purchase_request_lines" column_invisible="1" />
+                <button
+                    column_invisible="purchase_request_lines == False"
```
Comment (youring) [inline purchase_request/views/purchase_order_view.xml]:
```suggestion
                    invisible="not purchase_request_lines"
```

**Review by youring — CHANGES_REQUESTED** [review]
Runboat error:
`Caused by: EvalError: Can not evaluate python expression: (bool(purchase_request_lines == False))
    Error: Name 'purchase_request_lines' is not defined`
With fixes suggested by running [views_migration_17](https://github.com/OCA/server-tools/tree/17.0/views_migration_17)

**Comment by sonhd91** [comment]
> Runboat error: `Caused by: EvalError: Can not evaluate python expression: (bool(purchase_request_lines == False)) Error: Name 'purchase_request_lines' is not defined` With fixes suggested by running [views_migration_17](https://github.com/OCA/server-tools/tree/17.0/views_migration_17)

Thanks you, I just updated it, please check again

**Comment by youring** [comment]
Running smoothly now.
I tested the auto generate purchase request feature according to readme\CONFIGURE.md on runboat

1. Go to product form > Inventory tab, tick Buy and Purchase Request field
2. Create a new sale order containing that product, confirm the order.
3. There is no purchase request generated.

Maybe some steps I missed?

Thanks.

File: purchase_request/readme/CONFIGURE.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+To configure the product follow this steps:
+
+1.  Go to a product form.
+2.  Go to *Inventory* tab.
+3.  Check the box *Purchase Request* along with the route *Buy*.
```
Comment (youring) [inline purchase_request/readme/CONFIGURE.md]:
```suggestion
3.  Check the box *Purchase Request* along with the route *Buy* and *Replenish on Order (MTO)*
Note that MTO route is archived by default, you should unarchive it first: 
Go to Inventory > Configuration > Routes, in the Search dropdown menu, click *Archived* to show *Replenish on Order (MTO)* route and *Unarchive* it.
```

**Review by youring — CHANGES_REQUESTED** [review]
MTO route is Archived by default since v14. IMO it's a required step for auto generate purchase request from sale.

File: purchase_request/readme/CONFIGURE.md (hunk 4 above)
Comment (sonhd91) [reply purchase_request/readme/CONFIGURE.md]: Thanks you for suggestion, branch updated, please checked again

**Review by gurneyalex — APPROVED** [review]
Can you squash the translations commits by the same author, as per the guidelines?

preapproving

**Comment by sonhd91** [comment]
> Can you squash the translations commits by the same author, as per the guidelines?
> 
> preapproving

Hello, squash done, could you please take a look ?

**Pull request merged**